scipy/optimize/_nonlin.py

```diff
@@ -701,7 +701,7 @@ def __array__(self, dtype=None, copy=None):
 
     def collapse(self):
         """Collapse the low-rank matrix to a full-rank one."""
-        self.collapsed = np.array(self)
+        self.collapsed = np.array(self, copy=None)
```

Comment:
This doesn't look right just yet. On your branch, if I build against NumPy `main` and then downgrade to our lowest-supported runtime version before testing I'll get a bunch of failures like `ValueError: NoneType copy mode not allowed.`

I know we previously were using `scipy/_lib/_util.py ` `copy_if_needed` shims for stuff a bit like this.